ProductionSystem/Utilities/ProdValidator.py

```diff
@@ -136,9 +136,8 @@ def compareValues(self, value, parentValue):
     :param str parentValue: a meta data value (str, int, ...) to be compared with value
     """
     return set(
-        value.values()[0]).issubset(
+        str(value.values()[0])).issubset(
```

Comment:
according to the docstring, `value` is a `str`, but that is clearly not the case. What type are `value` and `parentValue`?

Reply:
You are right, value and parentValue can be str or int or float as checked also in:

https://github.com/DIRACGrid/DIRAC/blob/9a324b9406094b989fb64f90e2011ea301ea8f7d/ProductionSystem/Utilities/ProdValidator.py#L102

Should we simply change the doc string?

Reply:
`value.values()` for ints, floats or strings results in `AttributeError [...] no attribute 'values'`. 
Are `value` and `parentValue` dictionaries? Those have  a `values()` function.

Making a set out of a string gives a set of its characters.
```
<<< set(str({'foo': 'bar'}.values()[0]))
>>> {'a', 'b', 'r'}
```
Is that really the intent?

Reply:
`value.values()` for ints, floats or strings results in `AttributeError [...] no attribute 'values'`. 
Are `value` and `parentValue` dictionaries? Those have  a `values()` function.

Making a set out of a string gives a set of its characters.
```
<<< set(str({'foo': 'bar'}.values()[0]))
>>> {'a', 'b', 'r'}
```
Is that really the intent?

Reply:
Sorry you are right. These are dictionaries. What can be str, int or float or list of str, int and floats is:
value.values()[0]
and
parentValue.values()[0]

The fix is necessary because set doesn't work on int or float.

Reply:
But this cannot be what you want:
```
In [1]: set(str({'foo': 1000}.values()[0]))
Out[1]: {'0', '1'}
```
?

Reply:
You are absolutely right. 
This piece of code was correct indeed in the original version.
What was missing was some additional check and action in checkformatQuery.
See my last commit: 

90b68abb6eacab4a0b083e0c727dc8bd88c7eab6

Reply:
That makes sense!